cycles-ledger/src/lib.rs

```diff
@@ -177,6 +178,44 @@ pub fn transfer_from_error_to_transfer_error(e: TransferFromError) -> TransferEr
     }
 }
 
+// Traps if the error is InsufficientAllowance
+pub fn withdraw_from_error_to_withdraw_error(e: WithdrawFromError) -> WithdrawError {
+    match e {
+        WithdrawFromError::InsufficientFunds { balance } => {
+            WithdrawError::InsufficientFunds { balance }
+        }
+        WithdrawFromError::InsufficientAllowance { .. } => {
+            ic_cdk::trap("InsufficientAllowance error should not happen for withdraw")
```

Comment:
```suggestion
            ic_cdk::trap("InsufficientAllowance error should not happen for withdraw_from")
```

Reply:
Correct as written. `InsufficientAllowance` can't be turned into a `WithdrawError`